In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv('bestsellers with categories.csv')

# Display the first five rows
df.head()

In [ ]:
df.shape

Набір даних містить 550 книг з 7 змінними (стовпчиками) для кожної книги.

Дайте відповідь: Скільки книг містить набір даних?
Набір даних містить 550 книг.



In [ ]:

df.columns = ['name', 'author', 'user_rating', 'reviews', 'price', 'year', 'genre']
df.head()


# Первинне дослідження даних

### 1. Перевіре, чи у всіх рядків вистачає даних: виведіть кількість пропусків (na) у кожному зі стовпців

In [ ]:
missing_values = df.isna().sum()
missing_values

### Відповідь
1. Немає пропущених даних

### 2. Перевірте, які є унікальні значення в колонці `genre`

In [ ]:
unique_genres = df["genre"].unique()
unique_genres

### Відповідь
2. Є два унікальні жанри: Non Fiction, Fiction

### 3. Тепер подивіться на розподіл цін: побудуйте діаграму (використовуйте kind='hist')


In [ ]:
plt.figure(figsize=(10,6))
df["price"].plot(kind="hist", bins=30, edgecolor="black", color="skyblue")
plt.title("Розподіл цін")
plt.xlabel("Ціна ($)")
plt.ylabel("К-ть книжок")
plt.grid(True, which="both", linewidth=0.5)
plt.show()

### 4. Визначте, яка ціна у нас максимальна, мінімальна, середня, медіанна

In [22]:
max_price = df['price'].max()
min_price = df['price'].min()
mean_price = df['price'].mean()
median_price = df['price'].median()

max_price, min_price, mean_price, median_price

(105, 0, 13.1, 11.0)

### Відповідь

 4.1 Максимальна ціна?
    105
 4.2. Мінімальна ціна?
    0
 4.3. Середня ціна?
    13,1
 4.4 Медіанна ціна?
    11,0

# Пошук та сортування даних

### 1. Який рейтинг у датасеті найвищий?

In [24]:
highest_rating = df['user_rating'].max()
print(f"Відповідь: {highest_rating}")

Відповідь: 4.9


### 2.  Скільки книг мають такий рейтинг?

In [25]:
books_HR = df[df['user_rating'] == highest_rating].shape[0]
print(f"Відповідь: {books_HR}")

Відповідь: 52


### 3. Яка книга має найбільше відгуків?

In [28]:
book_MR = df[df['reviews'] == df['reviews'].max()]['name'].iloc[0]
print(f"Відповідь: Це книга - '{book_MR}'")

Відповідь: Це книга - 'Where the Crawdads Sing'


### 4.  З тих книг, що потрапили до Топ-50 у 2015 році, яка книга найдорожча

In [29]:
most_expensive_book_2015 = df[df["year"] == 2015].sort_values(by="price", ascending=False).iloc[0]["name"]
print(f"Відповідь: Це книга - '{most_expensive_book_2015}'")

Відповідь: Це книга - 'Publication Manual of the American Psychological Association, 6th Edition'


### 5. Cкільки книг жанру Fiction потрапили до Топ-50 у 2010 році


In [31]:
answer = df[(df['year'] == 2010) & (df['genre'] == 'Fiction')].shape[0]
print(f"Відповідь: {answer} книг.")

Відповідь: 20 книг.


### 6. Скільки книг з рейтингом 4.9 потрапило до рейтингу у 2010 та 2011 роках

In [43]:
answer6 = df[(df['user_rating'] == 4.9) & (df['year'].isin((2010, 2011)))].shape[0]
print(f"Відповідь: {answer6} книг.")

Відповідь: 1 книг.


### 7. Давайте відсортуємо за зростанням ціни всі книги, які потрапили до рейтингу в 2015 році і коштують дешевше за 8 доларів.  Яка книга остання у відсортованому списку?

In [45]:
answer7 = df[(df['year'] == 2015) & (df['price'] < 8)].sort_values(by='price').iloc[-1]['name']
print(f"Відповідь: Остання книга у списку - '{answer7}'")

Відповідь: Остання книга у списку - 'Old School (Diary of a Wimpy Kid #10)'
